src/output/shell_output.go

```diff
@@ -69,6 +69,16 @@ func MonitorState(state *core.BuildState, numThreads int, plainOutput, keepGoing
 	failedTargets := []core.BuildLabel{}
 	failedNonTests := []core.BuildLabel{}
 	for result := range state.Results {
+		if state.DebugTests && result.Status == core.TargetTesting {
+			stop <- struct{}{}
+			<-displayDone
```

Comment:
I am relying on the precondition, yeah. I still want to handle the following part to process results because that makes everything work out OK if the tests all pass.